src/charm.py

```diff
@@ -1594,6 +1604,18 @@ def get_available_resources(self) -> Tuple[int, int]:
 
         return cpu_cores, allocable_memory
 
+    def on_deployed_without_trust(self) -> None:
+        """Blocks the application and returns a specific error message for deployments made without --trust."""
+        self.unit.status = BlockedStatus(
+            "Unauthorized access to k8s resources. Is the app trusted? See logs"
+        )
```

Comment:
The blocked status should be removed when the application is trusted.

Reply:
During my tests, I saw that once the application gets properly trusted, a `config_changed` event gets triggered which removes the blocked status. In this case, is it necessary to explicitly unset it somewhere in the charm code?

Reply:
I ask this because I see in the charm code that we have some cases where we set the status to blocked but I didn't see it getting explicitly unset afterwards. Maybe I missed it though.

Reply:
As long as we don't get stuck with blocked status it should be fine. Down the road we should overhaul the way we set statuses, but that's out of scope of the PR.

Reply:
The intended behavior we want here will be defined and verified by the integration test I'll create shortly.